llvm/lib/Target/PowerPC/PPCReduceCRLogicals.cpp

```diff
@@ -481,6 +486,8 @@ PPCReduceCRLogicals::createCRLogicalOpInfo(MachineInstr &MIParam) {
       MachineInstr *Def2 = lookThroughCRCopy(MIParam.getOperand(2).getReg(),
                                              Ret.SubregDef2,
                                              Ret.CopyDefs.second);
+      if (Ret.SubregDef2 == 0)
+        Ret.SubregDef2 = MIParam.getOperand(2).getSubReg();
```

Comment:
If there are 2 subregisters, do you need to compose? I'd expect something like this to be unconditional or a compose